### PR TITLE
Improved averages graph

### PR DESCRIPTION
**Pull request by moreofmorris**
### Purpose
This is to improve some of the visual aspects of the averages graph. Nicer looking spacing, nicer looking columns at various date ranges, a more consistent tooltip and a small border to help with structure. This relates to this JIRA task: https://buffer.atlassian.net/browse/BAN-98

### Notes
I've added in some mock data to the story of the averages graph component and made some custom code for the tooltip to stay in the same y position when scrolling.

### Review
Be awesome if either @msanroman or @federicoweber could take a gander! 😄 

#### Staging Deployment

_This repo is CI/CD enabled and staging deployment should be available at:
https://task-improved-averages-graph-analyze.dev.buffer.com :smile:_

### CODE REVIEW

**Comment by moreofmorris** [comment]
Hey there @msanroman (and/or @federicoweber!) but I'm having trouble with the code coverage on this one. I don't really know how to get it to pass. I tried something late on Friday evening but that didn't work. I'd love to understand a little deeper how this cove coverage stuff works 🤔But any help on getting this pull request wrapped up would be SUPER helpful! Could you spare some time? 😄

File: packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx [hunk 1]
```diff
@@ -40,24 +40,33 @@ export default {
...
+    crosshairs: true,
     formatter() {
       const point = this.points[0].point;
       return reactDOM.renderToStaticMarkup(<ChartTooltip point={point} />);
     },
+    backgroundColor: '#343E46',
+    borderRadius: 4,
+    borderWidth: 0,
+    hideDelay: 10,
+    pointFormatter: () => `${this.series.name}: <b>${this.y}</b><br/>`,
```
Comment (federicoweber) [inline packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx]: I could be wrong but I think we can remove this @moreofmorris

File: packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx [hunk 2]
```diff
@@ -40,24 +40,33 @@ export default {
...
     shadow: false,
     useHTML: true,
-    positioner: () => ({ x: 0, y: 0 }),
-    style: {
-      pointerEvents: 'none',
+    positioner: function (boxWidth, boxHeight, point) { // eslint-disable-line
+      const chart = this.chart;
+      let x = (point.plotX + chart.plotLeft) - (boxWidth - 30);
```
Comment (federicoweber) [inline packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx]: great one @moreofmorris. 

To ease maintainability, would you mind assigning the `30` to a variable and maybe breaking down the rest of the calculation into named variables?

Maybe something like:
```
const boxOffset = 30; 
const origin = point.plotX + chart.plotLeft;
let x = origin - (boxWidth - boxOffset);

if (x < 0) x = 0;

const isOveflowing = (x +boxWidth) > chart.plotWidth;
if (isOverflowing) {
	const overflowingAmount = (x + boxWidth) - chart.plotWidth;
	x -= overflowingAmmount;
}

return { x, y: 0 };
```

Sorry for the poor names, I'm sure you can come up with better ones :smile:

File: packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx (hunk 1 above)
Comment (moreofmorris) [reply packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx]: I was trying this out locally... You're totally right Federico. The `pointFormatter` isn't required at all. But `borderRadius`, `borderWidth` and `hideDelay` still seem required. Thanks for pointing this out!

**Comment by moreofmorris** [comment]
Hey there @federicoweber, I made those changes but I'm still having some troubles with code coverage. I'm just not sure what I'm doing wrong here? 🤔

File: packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx (hunk 1 above)
Comment (federicoweber) [reply packages/shared-components/GridItem/components/GridItemChart/chartConfig.jsx]: :+1: yeah I was referring just to the `pointFormatter` @moreofmorris :smile:

**Comment by federicoweber** [comment]
Hey @moreofmorris, codecov is lamenting that the positioner it's not fully covered. We can add a few unit tests for that if we really want to, but I think this is good to ship :smile:. 

Thank you :smile:

**Comment by moreofmorris** [comment]
Thanks so much @federicoweber, I'll merge it in 👍Maybe I can reach out to you separately and ask a bit more about this code coverage stuff so I can learn better for the future 😄

**Comment by federicoweber** [comment]
yeah happy to help you with that @moreofmorris :smile:

**Pull request merged**